### PR TITLE
models: avoid shared dict in all instances of WorkflowObjectModel

### PR DESCRIPTION
**Pull request by jmartinm**
* Prevents all instances of WorkflowObjectModel from having the same
  reference to a dictionary in the data and extra_data columns. This
  was leading to unexpected behaviour.

Signed-off-by: Javier Martin Montull <javier.martin.montull@cern.ch>

### CODE REVIEW

**Comment by jacquerie** [comment]
Failure is caused by https://github.com/inveniosoftware/troubleshooting/issues/9. Since we need this, I think we need to bite the bullet and ignore/fix the docstrings ASAP.

**Comment by jacquerie** [comment]
This will still fail because of https://github.com/inveniosoftware/troubleshooting/issues/8, as seen in https://github.com/inveniosoftware-contrib/invenio-workflows/pull/39. The solution is apparently to pin `celery<4.0.0`.

File: run-tests.sh [hunk 1]
```diff
@@ -24,6 +24,6 @@
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
 pydocstyle invenio_workflows && \
-sphinx-build -qnNW docs docs/_build/html && \
+sphinx-build -qnN docs docs/_build/html && \
 python setup.py test && \
-sphinx-build -qnNW -b doctest docs docs/_build/doctest
+sphinx-build -qnN -b doctest docs docs/_build/doctest
```
Comment (david-caro) [inline run-tests.sh]: I'm not sure this is better than whitelisting the specific warning.

File: invenio_workflows/models.py [hunk 2]
```diff
@@ -153,7 +153,7 @@ class WorkflowObjectModel(db.Model):
             postgresql.JSON(none_as_null=True),
             'postgresql',
         ),
-        default={},
```
Comment (david-caro) [inline invenio_workflows/models.py]: I'm curious, can you just pass `dict` instead of a full lambda?

Comment (jmartinm) [inline invenio_workflows/models.py]: Will try and let you know out of curiosity

File: run-tests.sh (hunk 1 above)
Comment (jmartinm) [inline run-tests.sh]: By looking at the error (https://travis-ci.org/inveniosoftware-contrib/invenio-workflows/jobs/227859081#L677) I am not sure which is the code of the warning we should disable.

File: invenio_workflows/models.py (hunk 2 above)
Comment (jmartinm) [inline invenio_workflows/models.py]: Yes, it works as well. Do you think passing `dict` is more clear ?

Comment (david-caro) [inline invenio_workflows/models.py]: imo yes, though we are using lambda everywhere else, so better keep the style and refactor all at once later (whenever... not a big deal).

File: run-tests.sh (hunk 1 above)
Comment (david-caro) [inline run-tests.sh]: That error is due to the celery tasks issue that @jacquerie pointed out (inveniosoftware/troubleshooting#8), maybe we can pin celery only on the docs requirements.

Comment (david-caro) [inline run-tests.sh]: There's a bit of a mess with the deps, you might need to change the `setup.py` and `tox.ini` files.

Comment (david-caro) [inline run-tests.sh]: For the tox.ini, you have to put each requirement on a separate line (for what I could figure out):
```
    {docs,docstyle,doctests}: Sphinx>=1.5.1
    {docs,docstyle,doctests}: Celery<4.0
```

**Pull request closed without merging**